### PR TITLE
feat: add resolver contracts for Avalanche Benqi

### PR DESCRIPTION
**Pull request by babkendev01**
Add avalanche resolver, for Benqi:

Protocol: https://app.benqi.fi/
Docs: https://docs.benqi.fi/

### CODE REVIEW

File: contracts/protocols/avalanche/benqi/interfaces.sol [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+interface QiTokenInterface {
+    function exchangeRateStored() external view returns (uint256);
+
+    function borrowRatePerBlock() external view returns (uint256);
+
+    function supplyRatePerBlock() external view returns (uint256);
```
Comment (Vaibhav20201) [inline contracts/protocols/avalanche/benqi/interfaces.sol]:
``` suggestion
    function borrowRatePerTimestamp() external view returns (uint256);

    function supplyRatePerTimestamp() external view returns (uint256);

Comment (Vaibhav20201) [inline contracts/protocols/avalanche/benqi/interfaces.sol]: In the qitoken contract there are no functions named supplyRatePerBlock() or borrowRatePerBlock() (these are there in the ctoken contract). Instead, it has supplyRatePerTimestamp() and borrowRatePerTimestamp(). Changed this and deployed the resolver here:- 0xCd4df1f8dbd019258e9d26B49Abf93E961C5D2D6

File: contracts/protocols/avalanche/benqi/main.sol [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                priceInAVAX,
+                priceInUSD,
+                qiToken.exchangeRateStored(),
+                qiToken.balanceOf(owner),
+                qiToken.borrowBalanceStored(owner),
+                _totalBorrowed,
+                add(_totalBorrowed, qiToken.getCash()),
+                troller.borrowCaps(qiAddress[i]),
+                qiToken.supplyRatePerBlock(),
+                qiToken.borrowRatePerBlock(),
```
Comment (Vaibhav20201) [inline contracts/protocols/avalanche/benqi/main.sol]: Changed supplyRatePerBlock(), borrowRatePerBlock() to supplyRatePerTimestamp(), borrowRatePerTimestamp() here aswell.

Comment (babkendev01) [reply contracts/protocols/avalanche/benqi/main.sol]: > Changed supplyRatePerBlock(), borrowRatePerBlock() to supplyRatePerTimestamp(), borrowRatePerTimestamp() here aswell.

@Vaibhav20201 thank you for the catch. I've pushed the changes for that.

**Comment by KaymasJain** [comment]
Resolver address: [0xCd4df1f8dbd019258e9d26B49Abf93E961C5D2D6](https://avascan.info/blockchain/c/address/0xCd4df1f8dbd019258e9d26B49Abf93E961C5D2D6)

**Review by KaymasJain — APPROVED** [review]

File: contracts/protocols/avalanche/benqi/helpers.sol [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        uint256 tokenPriceInAvax;
+        uint256 tokenPriceInUsd;
+        uint256 exchangeRateStored;
+        uint256 balanceOfUser;
+        uint256 borrowBalanceStoredUser;
+        uint256 totalBorrows;
+        uint256 totalSupplied;
+        uint256 borrowCap;
+        uint256 supplyRatePerBlock;
+        uint256 borrowRatePerBlock;
```
Comment (Vaibhav20201) [inline contracts/protocols/avalanche/benqi/helpers.sol]:
```suggestion
        uint256 supplyRatePerTimestamp;
        uint256 borrowRatePerTimestamp;
```

Comment (babkendev01) [reply contracts/protocols/avalanche/benqi/helpers.sol]: > 

@Vaibhav20201 @KaymasJain 
updated this as well in the last commit.

**Review by KaymasJain — APPROVED** [review]

**Pull request merged**